midend/interpreter.cpp

```diff
@@ -585,6 +585,56 @@ SymbolicVoid* SymbolicVoid::instance = new SymbolicVoid();
 
 /*****************************************************************************************/
 
+void ExpressionEvaluator::postorder(const IR::Operation_Ternary* expression) {
+    auto e0 = get(expression->e0);
+    if (e0->is<SymbolicError>()) {
+        set(expression, e0);
+        return;
+    }
+    auto e1 = get(expression->e1);
+    if (e1->is<SymbolicError>()) {
+        set(expression, e1);
+        return;
+    }
+    auto e2 = get(expression->e2);
+    if (e2->is<SymbolicError>()) {
+        set(expression, e2);
+        return;
+    }
+    auto clone = expression->clone();
+    BUG_CHECK(e0->is<SymbolicInteger>(), "%1%: expected an SymbolicInteger", e0);
+    BUG_CHECK(e1->is<SymbolicInteger>(), "%1%: expected an SymbolicInteger", e1);
```

Comment:
Why should e1 be an integer? Any other type would work here.

Reply:
Looks like this follows the implementation of IR::Operation_Binary below. Why wasn't SymbolicInteger an issue before?

Reply:
There aren't many binary operations on non-integers. Comparisons are the only ones I can think of.
There are some passes in the midend which eliminate structure comparisons into field-wise comparisons.
So perhaps that's why. I don't think we eliminate ternary operations as well, but maybe we should.
Anyway, this deserves a test.

Reply:
> Looks like this follows the implementation of IR::Operation_Binary below. Why wasn't SymbolicInteger an issue before?

I think that this class was used for integer calculations only, but [this](https://github.com/p4lang/p4c/blob/main/frontends/common/constantFolding.h) class is used for calculation and it supports integer and booleans.  So, I have tried to add boolean support

Reply:
> There aren't many binary operations on non-integers. Comparisons are the only ones I can think of.
> There are some passes in the midend which eliminate structure comparisons into field-wise comparisons.
> So perhaps that's why. I don't think we eliminate ternary operations as well, but maybe we should.
> Anyway, this deserves a test.

I think that my last changes help to increase the functionality of this class. Now, It makes an evaluation of the boolean expressions.